var/spack/repos/builtin/packages/python/package.py

```diff
@@ -64,21 +67,36 @@ def install(self, spec, prefix):
         # Rest of install is pretty standard except setup.py needs to
         # be able to read the CPPFLAGS and LDFLAGS as it scans for the
         # library and headers to build
-        configure_args= [
-                  "--prefix=%s" % prefix,
-                  "--with-threads",
-                  "--enable-shared",
-                  "CPPFLAGS=-I%s/include -I%s/include -I%s/include -I%s/include -I%s/include -I%s/include" % (
+        configure_args = []
+        configure_args.extend(['--prefix=%s' % prefix,
+                               '--with-threads',
+                               '--enable-shared'])
+
+        cppflags = '-I%s/include -I%s/include -I%s/include -I%s/include' % (
                        spec['openssl'].prefix, spec['bzip2'].prefix,
-                       spec['readline'].prefix, spec['ncurses'].prefix,
-                       spec['sqlite'].prefix, spec['zlib'].prefix),
-                  "LDFLAGS=-L%s/lib -L%s/lib -L%s/lib -L%s/lib -L%s/lib -L%s/lib" % (
+            spec['readline'].prefix, spec['ncurses'].prefix)
+        cppflags += ' -I%s/include -I%s/include' % (
+            spec['sqlite'].prefix, spec['zlib'].prefix)
+        ldflags = '-L%s/lib -L%s/lib -L%s/lib -L%s/lib' % (
                        spec['openssl'].prefix, spec['bzip2'].prefix,
-                       spec['readline'].prefix, spec['ncurses'].prefix,
+            spec['readline'].prefix, spec['ncurses'].prefix)
+        ldflags += ' -L%s/lib -L%s/lib' % (
                        spec['sqlite'].prefix, spec['zlib'].prefix)
-                  ]
+        if '+tk' in spec:
+            configure_args.extend([
+                '--with-tcltk-includes=-I%s/include -I%s/include' % (
+                    spec['tk'].prefix, spec['tcl'].prefix),
+                '--with-tcltk-libs=%s/lib %s/lib' % (
+                    spec['tk'].prefix, spec['tcl'].prefix)])
+            cppflags += ' -I%s/include -I%s/include' % (
+                spec['tk'].prefix, spec['tcl'].prefix)
```

Comment:
@mathstuf I though I had tried the other two permutations (without cppflags/ldflags or without the tcltk options). However, your question inspired me to go back and try the different combinations again. To my surprise, I only needed the cppflags and ldflags to obtain a working Tkinter package.

If I eliminate the cppflags and ldflags and only use the tcltk configure options, the build completes successfully, but the Tkinter module is broken.  

I will update the PR shortly with an updated version that removes the tcltk options.